### PR TITLE
Self-Gorillafication Nerf

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

So, I buffed gorillas a while back for traitors to get more mileage out of them, and outside of needing to adjust that one wizard spell I've been overall pleased with the results. However, while I knew that Geneticists were capable of turning themselves into gorillas without the magilitis injector, I thought it came with great difficulty but hoo boy I was wrong. Turns out, its quite easy to turn yourself into a gorilla with the three roundstart genetics setups you start with, and all you need a little bit of toxin-healing chems (which don't really have a downside and can stack!) or a medibot to pull it off. People have been using this to validhunt as the newly buffed gorillas as early as 7 minutes or less into the round, which I think sucks! So I'm here to change it.

If you become a gorilla from genetics, you become a new subtype with nerfed stats. The stats are as follows:

180 health (from 220)
Melee damage 15-18 (from 25-30)
Obj Damage 25 (from 40)
Speed 0.1 (from -0.1)
Paralyze chance 0 (from 20)
No limb ripping

## Why It's Good For The Game

It's very easy to become a gorilla right now with little to no effort and to great effect, seeing as gorillas are pretty strong now. This brings crew gorillas back in line to before gorillas got buffed for traitor usage, which should ideally make this more balanced overall.

## Changelog

:cl:
balance: Gorillas made from giving monkeys genetic damage are weaker than their normal counterparts.
/:cl:

### CODE REVIEW

**Comment by Xkallubar** [comment]
Man, I have to set up 10 genetics setups in a row, get clever and telekinesis, and spam genetic damage for a whole hour before becoming a gorilla. Now it's going to take even longer.

**Comment by Vect0r2** [comment]
The techweb changes that puts genetics tech behind xenobio & botany have made it so it's extremely hard to get a ton of genetic damage quickly.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Just make a slightly weaker version of the gorilla for the self-gorillafication. Don't make it a total pain in the ass to get to begin with, because there's no fun in that.

**Comment by IndieanaJones** [comment]
Ok, so here's the problem. There's nothing stopping you from making a bunch of genetic scanners right now to achieve gorillafication in extremely short time (every use gives 100 - 250 genetic damage, cooldown is 60 seconds per machine), they're very easy to make and you work in the department where you can get everything you need to build them. Its so easy though right now you don't even need extra to pull it off.

If we make it transform you into a weaker gorilla, we have this weird moment where there's already a weaker gorilla in the code base, but it's that much weaker that nobody would want to turn into it. So we either then make a new gorilla subtype for this one instance and have two weaker gorilla subtypes, or we varedit the resulting gorilla from gorillize() to achieve our desired results.

Thoughts?

PS. For the record, I don't think having people build a bunch of genetic consoles to pull this off is bad, considering if you're doing this, this is probably the only project you're going to pull off this shift. They already build them anyway because it makes this already short process on live even shorter and DNA consoles are cheap.

**Comment by BlueMemesauce** [comment]
Currently, gorillafication is so fast that I don't think you could even research and build more DNA scanners in that time. If it was made slower, people probably would build more to speed up the process.

**Comment by moocowswag** [comment]
This was already nerfed , you keep saying a geneticist can easily become a gorilla but if it was then the previous set of double nerfs that both oranges and timbertoes oversighted had failed this whole time

A long endeavor/process is not easily, 5 minutes is more than 10% of the avg round and we dont even have an avg time to become gorilla from 0 genetic damage.

**Comment by IndieanaJones** [comment]
> This was already nerfed , you keep saying a geneticist can easily become a gorilla but if it was then the previous set of double nerfs that both oranges and timbertoes oversighted had failed this whole time
> 
> A long endeavor/process is not easily, 5 minutes is more than 10% of the avg round and we dont even have an avg time to become gorilla from 0 genetic damage and still needing to gather anti toxin
> 
> the 400 seconds a previous comment estimated alone is over 6 mins and thats after instability is reached
> 
> and building extra scanners, whats wrong with that? Its a further resource and research investment.

Yeah, it is incredibly easy for people to currently become gorillas and none of the previous nerfs actually stunted how quickly you could pull this off at all. I do not consider 5 - 8 minutes to be a long process in the slightest, especially considering how strong it is and how easily a crew gorilla can molleywhop most antagonists in a 1v1 with no problem at all.

I personally had no problems with creating more scanners, and ideally I'd prefer people have to build them to get a good reward with effort as opposed to nerfing the gorilla granted by genetics outright, thus giving a medicore reward for medicore effort. I'd basically be a soft removal as opposed to enforcing a good amount of effort go into a good reward.

However, that's what I'm going to do now.

**Comment by IndieanaJones** [comment]
The PR has been changed entirely. Please read the new description.

**Comment by zoomachina** [comment]
Hold on, are you saying it's extremely easy to become gorilla because you've seen it happen or because you did some numbers in your head and figured it is?

**Comment by ATH1909** [comment]
> EDIT: Genetics Gorilla stats:
> 
> * 180 health (from 220)
> * Melee damage 15-18 (from 25-30)
> * Obj Damage 25 (from 40)
> * Speed 0.1 (from -0.1)
> * Paralyze chance 0 (from 20)
> * No limb ripping

Don't these new stats, uh, suck? Why would a geneticist ever want to do this? You're slower than your prey and your meaty gorilla fists are... about as effective as a cleaver. Am I missing something here?

**Comment by IndieanaJones** [comment]
> Hold on, are you saying it's extremely easy to become gorilla because you've seen it happen or because you did some numbers in your head and figured it is?

I've seen it and done it myself. It really isn't hard to procure some anti-toxin meds and walk between the genetic consoles netting genetic damage.

**Comment by excalisword** [comment]
They also have health regeneration.

**Comment by Eronymun** [comment]
There needs to be some sort of visual distinction between these weaker gorillas and the stronger kind, conveying that kind of info is good for gameplay. Maybe have the weaker gorillas be like orangutans instead of silverbacks?

**Comment by IndieanaJones** [comment]
> There needs to be some sort of visual distinction between these weaker gorillas and the stronger kind, conveying that kind of info is good for gameplay. Maybe have the weaker gorillas be like orangutans instead of silverbacks?

They have different names, descriptions, and are slightly smaller (0.9x scale)

**Comment by IndieanaJones** [comment]
> Do the nerfs make them identical to prebuffed gorillas? If so thats fair

The health and melee damage is lower than prebuff gorillas, along with losing paralyze chance, but object damage and movement speed is higher. I imagine speed is the big one here, as having more health or dealing slightly more damage per swing doesn't matter if you're hopelessly outpaced to begin with.

**Comment by Xackii** [comment]
Why should someone spend 2 hours of a round to become a basic mob that will die from laser spam? 
Sounds like a useless mechanic that just exists

**Comment by IndieanaJones** [comment]
> Why should someone spend 2 hours of a round to become a basic mob that will die from laser spam? Sounds like a useless mechanic that just exists

It takes like 5 to 8 minutes to do become a gorilla

**Comment by Xackii** [comment]
> It takes like 5 to 8 minutes to become a gorilla

This is the first time I’ve heard about such a mechanic and I still don’t understand how to do it.

**Comment by Ravellon** [comment]
The wiki says that it's 25% chance to become gorilla once you hit 2500 genetic damage.
The wiki is wrong. In actuality it's 0.25% and it has been this way since the beginning. I have no idea how you can become a gorilla in 5 minutes without unrealistic amount of luck.

**Comment by BlueMemesauce** [comment]
> The wiki says that it's 25% chance to become gorilla once you hit 2500 genetic damage. The wiki is wrong. In actuality it's 0.25% and it has been this way since the beginning. I have no idea how you can become a gorilla in 5 minutes without unrealistic amount of luck.
> 
> Which also tells me this is not a problem that exists in the game currently, this is purely theoretical problem.

According to this calculator, it's a 52.5% chance to become a gorilla within 5 minutes

image
https://www.omnicalculator.com/statistics/binomial-distribution

**Comment by Ravellon** [comment]
Then something else is going on because on local I had been waiting for an hour and no gorillafication

**Comment by Xackii** [comment]
@IndieanaJones Maybe you should remove the randomness in getting the gorilla and make the method of getting it more interesting? Because this all sounds like a forgotten mechanic that 1/1000 people will do because they have nothing to do.

**Comment by ATH1909** [comment]
> with how they are right now gorillas are both fun and strong enough to not be just a gimmick which imo is not a bad thing considering once you turn into one you cant revert to human (and i believe you cant communicate in human language anymore), you get round removed if you die and it is pretty tedious to become one. i dont think this change is needed at all.

you don't get round-removed when you die- someone can lazarus injector you, strange reagent you, or butcher your corpse to generate a brain that can then be shoved into a human body.

File: code/modules/mob/living/basic/farm_animals/gorilla/gorilla.dm [hunk 1]
```diff
@@ -167,4 +167,21 @@
...
+	desc = "A gorilla created via \"advanced genetic science\". While not quite as strong as their wildborne brethren, this simian still packs a punch."
+	melee_damage_lower = 15
+	melee_damage_upper = 18
+	obj_damage = 25
+	speed = 0.1
+	paralyze_chance = 0
+
+/mob/living/basic/gorilla/genetics/Initialize(mapload)
+	. = ..()
+	transform *= 0.9
```
Comment (Ghommie) [inline code/modules/mob/living/basic/farm_animals/gorilla/gorilla.dm:184]:
```suggestion
```

File: code/modules/mob/living/basic/farm_animals/gorilla/gorilla.dm [hunk 2]
```diff
@@ -167,4 +167,21 @@
...
+/mob/living/basic/gorilla/genetics
+	name = "Lab Gorilla"
+	maxHealth = 180
+	health = 180
+	desc = "A gorilla created via \"advanced genetic science\". While not quite as strong as their wildborne brethren, this simian still packs a punch."
+	melee_damage_lower = 15
+	melee_damage_upper = 18
+	obj_damage = 25
+	speed = 0.1
+	paralyze_chance = 0
```
Comment (Ghommie) [inline code/modules/mob/living/basic/farm_animals/gorilla/gorilla.dm:180]:
```suggestion
	paralyze_chance = 0
	current_size = 0.9
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**